lib/ramble/ramble/software_environments.py

```diff
@@ -342,17 +342,22 @@ def mapped_environments(self, raw_env):
             yield env
 
     def specs_equiv(self, spec1, spec2):
-        all_keys = set(spec1.keys())
-        all_keys.update(set(spec2.keys()))
+        all_keys = set()
+        count1 = 0
+        count2 = 0
 
-        if len(all_keys) != len(spec1.keys()):
-            return False
+        for key in spec1.keys():
+            if spec1[key]:
```

Comment:
Are you trying to check "None"ness or "truthy"ness here? If it's None I think `if spec1[key] is not None:` is preferred 

Functionally it doesn't matter but I guess as written we would not check where one key=0 compared against key=None

Reply:
Good catch. I'm testing None-ness. This will be updated in a commit once I push an update.